python/lsst/ip/isr/binExposureTask.py

```diff
@@ -0,0 +1,202 @@
+            )
+        if config and config.exposureStorageClass != self.inputExposure.storageClass:
+            self.inputExposure = cT.Input(
+                name=self.inputExposure.name,
+                doc=self.inputExposure.doc,
+                storageClass=config.exposureStorageClass,
+                dimensions=self.inputExposure.dimensions,
+            )
+            self.binnedExposure = cT.Output(
+                name=self.binnedExposure.name,
+                doc=self.binnedExposure.doc,
+                storageClass=config.exposureStorageClass,
+                dimensions=self.binnedExposure.dimensions,
+            )
+
+
+class BinExposureConfig(
+    pipeBase.PipelineTaskConfig,
+    pipelineConnections=BinExposureConnections
+):
+    """Config for BinExposureTask"""
+    exposureDimensions = pexConfig.ListField(
+        # Sort to ensure default order is consistent between runs
+        default=sorted(BinExposureConnections.dimensions),
+        dtype=str,
+        doc="Override for the dimensions of the input and binned exposures.",
+    )
+    exposureStorageClass = pexConfig.Field(
+        default='ExposureF',
+        dtype=str,
+        doc="Override the storageClass of the input and binned exposures."
+    )
+    binFactor = pexConfig.Field(
+        dtype=int,
+        doc="Binning factor applied to both spatial dimensions.",
+        default=8,
+        check=lambda x: x > 1,
+    )
+
+
+class BinExposureTask(pipeBase.PipelineTask):
+    """Perform an nxn binning of an Exposure dataset type.
+
+    The binning factor is the same in both spatial dimensions (i.e.,
+    an nxn binning is performed). Each of the input Exposure's image
+    arrays are binned by the same factor.
+    """
+    ConfigClass = BinExposureConfig
+    _DefaultName = "binExposure"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        outputs = self.run(**inputs)
+        butlerQC.put(outputs, outputRefs)
+
+    @timeMethod
+    def run(self, inputExposure, binFactor=None):
+        """Perform an nxn binning of an Exposure.
+
+        Parameters:
+        -----------
+        inputExposure : `lsst.afw.image.Exposure` or one of its
+                        sub-types.
+            Exposure to spatially bin
+        binFactor : `int`, optional.
+            nxn binning factor. If not provided then self.config.binFactor
+            is used.
+
+        Returns:
+        --------
+        result : `lsst.pipe.base.Struct`
+            Results as a struct with attributes:
+
+            ``binnedExposure``
+               Binned exposure (`lsst.afw.image.Exposure` or one of its
+               sub-types. The type matches that of the inputExposure).
+        """
+        if not binFactor:
+            binFactor = self.config.binFactor
+        return pipeBase.Struct(
+            binnedExposure=binExposure(inputExposure, binFactor)
+        )
+
+
+def binExposure(inputExposure, binFactor=8):
```

Comment:
Can this method have the `'` switched to `"`?  I'm terrible at consistency myself, but doing it now will save some future clean-up ticket a file.